src/lib.rs

```diff
@@ -10,9 +10,11 @@
 mod address;
 pub mod bundle;
 mod circuit;
+mod constants;
 pub mod keys;
 mod note;
 pub mod primitives;
+mod spec;
```

Comment:
An idea: the `spec` module is full of a lot of stuff, what if it was broken up approximately by section in the spec, such as `prfs`, the functions only used for key derivation in `keys` such as `to_base`/`to_scalar` etc, `commit_ivk` with `simsemilla` _maybe_, etc. I kept thinking about doing something similar in `zebra-chain` for the prfs but never did. I'm mostly worried about a `spec` module becoming a `utils` or `toolbox ` or some other catchall/dumping ground.

Reply:
Heh, this was originally a `util.rs` file, but with far less in it! I decided to extract more small functions to here and rename it, in order to make the key component derivation visibly closer to the spec.

I'm ambivalent on adding module structure below this. I don't anticipate this file growing much larger, and I'm not really a fan of single-function modules, but I'm not strongly against the idea.

Reply:
Would there be a single-function module? I don't want that either